internal/httpserve/handlers/ent.go

```diff
@@ -501,3 +500,86 @@ func (h *Handler) getOrgByID(ctx context.Context, id string) (*ent.Organization,
 
 	return org, nil
 }
+
+// getOrgSettingByOrgID returns the organization settings from an organization ID and context
+func (h *Handler) getOrgSettingByOrgID(ctx context.Context, orgID string) (*ent.OrganizationSetting, error) {
+	orgGetCtx := privacy.DecisionContext(ctx, privacy.Allow)
+
+	org, err := h.getOrgByID(orgGetCtx, orgID)
+	if err != nil {
+		log.Error().Err(err).Msg("error retrieving organization")
+
+		return nil, err
+	}
+
+	if org.PersonalOrg {
+		log.Error().Msg("personal orgs do not have billing information")
+
+		return nil, ErrPersonalOrgsNoBilling
+	}
+
+	setting, err := org.Setting(ctx)
+	if err != nil {
+		log.Error().Err(err).Msg("error retrieving organization setting")
+
+		return nil, err
+	}
+
+	return setting, nil
+}
+
+func (h *Handler) fetchOrCreateStripe(context context.Context, orgsetting *ent.OrganizationSetting) (*stripe.Customer, error) {
```

Comment:
this function doesn't really belong in this file, this file is reusable queries to the database, but this is specific to the `checkout` handler (I think?) so it should probably live in there instead.